SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverRPC.cpp

```diff
@@ -382,7 +382,7 @@ bool FSpatialNetDriverRPC::ApplyRPC(Worker_EntityId EntityId, const FRPCPayload&
 	{
 		const TWeakObjectPtr<UObject> ActorReceivingRPC = NetDriver.PackageMap->GetObjectFromEntityId(EntityId);
 		AActor* Actor = CastChecked<AActor>(ActorReceivingRPC.Get());
```

Comment:
I would be tempted to remove this hard `checkf` instead and return `false` here? Again not totally sure what the outcome of this change would be.  `NetDriver.PackageMap->GetObjectFromEntityId(EntityId)` does not seem like a cheap operation to do for every payload (as you mention).

Reply:
I would agree. CanExtractRPC does perform that operation, of extracting an Actor from an EntityId.
Moving the check for each payload guarantees the worst possible outcome perf-wise.
If we wanted to be smarter, CanExtractRPC would return the actor pointer, pass it to ApplyRPC, which would check IsActorValid (Even though we asked for the actor's destruction, it would not be GCed immediately).
But the Receiver is Unreal-Agnostic, hence sliding an Actor pointer would be inelegant.
So keeping the CanExtractRPC before iterating on payloads, and silently failing subsequent ApplyRPC is the next best thing.